### PR TITLE
Application: use startup and dbus_register instead of activate

### PR DESCRIPTION
**Pull request by Marukesu**
- set the `ApplicationFlags.IS_SERVICE` flag so that glib don't try to call activate when running from terminal.

- move gtk and canberra configuration to `GLib.Application.startup ()`, and use
`GLib.Application.application_id` and `GLib.Application.resource_base_path` so we reduce places with an hardcoded app id.

- do all the dbus handing on `GLib.Application.dbus_register`, and only hold when we aquire the `org.freedesktop.Notifications` name.

### CODE REVIEW

**Comment by Marukesu** [comment]
@jeremypw i think this also supress the error, and are more inline with how GLib.Application/Gtk.Application works.

**Comment by jeremypw** [comment]
> @jeremypw i think this also supress the error, and are more inline with how GLib.Application/Gtk.Application works.

I'll bow to your greater knowledge :smile:

File: src/Application.vala [hunk 1]
```diff
@@ -25,41 +25,54 @@ public class Notifications.Application : Gtk.Application {
...
 
-    static construct {
+    protected override bool dbus_register (DBusConnection connection, string object_path) throws Error {
+        var server = new Notifications.Server ();
+
+        try {
+            connection.register_object ("/org/freedesktop/Notifications", server);
+        } catch (Error e) {
+            warning ("Registring notification server failed: %s", e.message);
+            return false;
```
Comment (tintou) [inline src/Application.vala:39]: Maybe:
```suggestion
            throw e;
```

**Review by tintou — APPROVED** [review]
Looks good to me :smiley: thanks for this

**Pull request merged**